.github/workflows/release.main.kts

```diff
@@ -0,0 +1,134 @@
+val SIGNING_KEY_ID by Contexts.secrets
+val SIGNING_KEY by Contexts.secrets
+val SIGNING_PASSWORD by Contexts.secrets
+
+workflow(
+    name = "Publish release to Maven Central",
+    on = listOf(
+        WorkflowDispatch()
+    ),
+    sourceFile = __FILE__,
+) {
+    val stagingRepoJob =
+        job(
+            id = "create-staging-repo",
+            name = "Create staging repository",
+            runsOn = RunnerType.UbuntuLatest,
+            outputs = object : JobOutputs() {
+                var repositoryId: String by output()
+            }
+        ) {
+            val createRepo = uses(
+                action = CreateNexusStagingRepo(
+                    username = expr { SONATYPE_USERNAME },
+                    password = expr { SONATYPE_PASSWORD },
+                    stagingProfileId = expr { SONATYPE_STAGING_PROFILE_ID },
+                )
+            )
+            jobOutputs.repositoryId = createRepo.outputs.repositoryId
+        }
+    val publishJob =
+        job(
+            id = "publish-artifacts",
+            runsOn = RunnerType.MacOSLatest,
+            needs = listOf(stagingRepoJob),
+        ) {
+            uses(action = Checkout())
+            uses(
+                name = "Set up JDK",
+                action = SetupJava(
+                    javaVersion = "11",
+                    distribution = SetupJava.Distribution.Zulu,
+                    cache = SetupJava.BuildPlatform.Gradle,
+                ),
+            )
+            uses(
+                name = "Cache Kotlin Konan",
+                action = Cache(
+                    path = listOf(
+                        "~/.konan/**/*",
+                    ),
+                    key = "kotlin-konan-${expr { runner.os }}",
+                ),
+            )
+            uses(
+                name = "Set up Gradle",
+                action = ActionsSetupGradle(
+                    gradleVersion = "wrapper",
+                ),
+            )
+            run(
+                name = "Publish",
+                command = "./gradlew publishAllPublicationsToSonatypeReleaseRepository",
+                env =
+                mapOf(
+                    "ORG_GRADLE_PROJECT_snake-kmp.ossrhUsername" to expr { SONATYPE_USERNAME },
+                    "ORG_GRADLE_PROJECT_snake-kmp.ossrhPassword" to expr { SONATYPE_PASSWORD },
+                    "ORG_GRADLE_PROJECT_snake-kmp.ossrhStagingRepositoryId" to expr { stagingRepoJob.outputs.repositoryId },
+                    "ORG_GRADLE_PROJECT_snake-kmp.signing.keyId" to expr { SIGNING_KEY_ID },
+                    "ORG_GRADLE_PROJECT_snake-kmp.signing.key" to expr { SIGNING_KEY },
+                    "ORG_GRADLE_PROJECT_snake-kmp.signing.password" to expr { SIGNING_PASSWORD },
+                )
+            )
+        }
+
+    job(
+        id = "close-staging-repo",
+        runsOn = RunnerType.UbuntuLatest,
+        condition = expr { publishJob.result.eq(AbstractResult.Status.Success) },
```

Comment:
```suggestion
        condition = expr { publishJob.result.eq(AbstractResult.Status.Success) },
```

Question for @krzema12: It's a little awkward reading `.eq` (and `.neq` below). Could `==` and `!=` be used? Or does something interfere with it?

Reply:
This is not a question to me, but I will try to share what I know: `==` and `!=` delegates to `equals` call. The signature of `equals` method is `fun equals(other: Any?): Boolean`. So, we cannot return a `String` from the method. Therefore, we cannot use it in `expr` function

Reply:
Like @OptimumCode wrote: we want to build a string here. GitHub Actions are very "stringy". I'm working on being able to specify such conditions in Kotlin, but we're not there yet, and it will for sure be more costly performance-wise.